### PR TITLE
refactor: Move GroupStatus update from responder to services (DEV-3292)

### PR DESCRIPTION
**Pull request by mpro7**
### Task Description/Number

Apart of moving update method also have been added:
- tests for `GroupService`,
- `KnoraGroupRepoLiveSpec` missing members implementation,
- methods that allow to cope with `KnoraGroup` instead of `Group` in `KnoraUserService`,
- `findByGroupMembership` method in `KnoraUserRepoLive`

Issue Number: DEV-

### PR Type

- [ ] build/chore: maintenance tasks (no production code change)
- [ ] docs: documentation changes (no production code change)
- [ ] feat: represents new features
- [ ] fix: represents bug fixes
- [ ] perf: performance improvements
- [x] refactor: represents production code refactoring
- [ ] test: adding or refactoring tests (no production code change)
- [ ] deprecated: Deprecation warning (ideally referencing a migration guide)

### Basic requirements for bug fixes and features

- [ ] Tests for the changes have been added
- [ ] Docs have been added / updated

### Does this PR introduce a breaking change?

- [ ] Yes

### Does this PR change client-test-data?

- [ ] Yes

### CODE REVIEW

**Review by siers — APPROVED** [review]

File: webapi/src/main/scala/org/knora/webapi/slice/admin/domain/service/KnoraGroupService.scala [hunk 1]
```diff
@@ -10,17 +10,20 @@ import zio.Task
 import zio.ZIO
 import zio.ZLayer
 
+import dsp.errors.BadRequestException
```
Comment (seakayone) [inline webapi/src/main/scala/org/knora/webapi/slice/admin/domain/service/KnoraGroupService.scala:13]: **issue**: `BadRequest` is an exception which should not be used in the domain package.

File: webapi/src/main/scala/org/knora/webapi/slice/admin/domain/service/KnoraUserService.scala [hunk 2]
```diff
@@ -138,10 +142,20 @@ case class KnoraUserService(
...
+           .unless(user.isInGroup.contains(group.groupIri))
     user <- updateUser(user, UserChangeRequest(groups = Some(user.isInGroup.filterNot(_ == group.groupIri)))).orDie
   } yield user
 
+  def removeUserFromKnoraGroup(user: User, group: KnoraGroup): IO[UserServiceError, KnoraUser] =
+    userRepo.findById(user.userIri).someOrFailException.orDie.flatMap(removeUserFromKnoraGroup(_, group))
+
+  def removeUserFromKnoraGroup(user: KnoraUser, group: KnoraGroup): IO[UserServiceError, KnoraUser] = for {
+    _ <- ZIO
+           .fail(UserServiceError(s"User ${user.id.value} is not member of group ${group.id.value}."))
```
Comment (seakayone) [inline webapi/src/main/scala/org/knora/webapi/slice/admin/domain/service/KnoraUserService.scala:154]: **question**: Why do we have to fail here?

If a user is not in the group the intention of the caller is fulfilled already so the effect could simple succeed.

**Review by seakayone — APPROVED** [review]

File: webapi/src/main/scala/org/knora/webapi/slice/admin/domain/service/KnoraGroupService.scala (hunk 1 above)
Comment (mpro7) [reply webapi/src/main/scala/org/knora/webapi/slice/admin/domain/service/KnoraGroupService.scala]: Changes in [d1d97fc](https://github.com/dasch-swiss/dsp-api/pull/3195/commits/d1d97fc1a9aeaba6118f0c7be088207085d24ff5)

File: webapi/src/main/scala/org/knora/webapi/slice/admin/domain/service/KnoraUserService.scala (hunk 2 above)
Comment (mpro7) [reply webapi/src/main/scala/org/knora/webapi/slice/admin/domain/service/KnoraUserService.scala]: [d1d97fc](https://github.com/dasch-swiss/dsp-api/pull/3195/commits/d1d97fc1a9aeaba6118f0c7be088207085d24ff5)

**Review by seakayone — APPROVED** [review]

**Pull request merged**